scripts/new_project.kts

```diff
@@ -9,23 +9,67 @@ object NewProject {
     private const val TEMPLATE_FOLDER_NAME = "CoroutineTemplate"
     private const val TEMPLATE_PACKAGE_NAME = "co.nimblehq.coroutine"
 
+    private const val APP_PATTERN = "^[A-Z\$][a-zA-Z0-9\$]*\$"
+    private const val PACKAGE_PATTERN = "^[a-z]+(\\.[a-z][a-z0-9]*)*\$"
+
     private val modules = listOf("app", "data", "domain")
 
     private var appName = ""
-    private val appNameWithoutSpace: String
-        get() = appName.replace(" ", "")
     private var packageName = ""
+
     private val projectPath: String
-        get() = rootPath + appNameWithoutSpace
+        get() = rootPath + appName
+
     private val rootPath: String
         get() = System.getProperty("user.dir").replace("scripts", "")
 
     fun generate(args: Array<String>) {
         handleArguments(args)
-        initializeNewProjectFolder()
-        cleanNewProjectFolder()
-        renamePackageNameFolders()
-        renamePackageNameWithinFiles()
+    }
+
+    private fun handleArguments(args: Array<String>) {
+        val agrumentError = "ERROR: Invalid Agrument name: Ensure define argruments => app-name={\"MyProject\"} package-name={com.sample.myproject}"
+        when (args.size) {
+            1 -> when {
+                args.first().startsWith(KEY_APP_NAME) -> showMessage("ERROR: No package has been provided")
+                args.first().startsWith(KEY_PACKAGE_NAME) -> showMessage("ERROR: No app name has been provided")
+                else -> showMessage(agrumentError)
+            }
+            2 -> args.map { arg ->
+                val (key, value) = arg.split(ARGUMENT_DELIMITER)
+                when (key) {
+                    KEY_APP_NAME -> validAppName(value)
+                    KEY_PACKAGE_NAME -> validPackageName(value)
+                    else -> showMessage(agrumentError)
+                }.also { executeNextSteps() }
+            }
+            else -> showMessage("ERROR: Require app-name and package-name to initialize the new project")
+        }
+    }
+
+    private fun validAppName(value: String) {
+        if (APP_PATTERN.toRegex().containsMatchIn(value)) {
+            appName = value
+        } else {
+            showMessage("ERROR: Invalid App Name: $value (needs to follow standard pattern {AppName})")
+        }
+    }
+
+    private fun validPackageName(value: String) {
+        if (PACKAGE_PATTERN.toRegex().containsMatchIn(value)) {
+            packageName = value
+        } else {
+            showMessage("ERROR: Invalid Package Name: $value (needs to follow standard pattern {com.example.package})")
```

Comment:
Ditto here

Reply:
`validPackageName` doesn't show the error if we run like `package-name=comtest.mytest` 🙏

Reply:
If I'm not mistaken, a package name with two packages is allowed in the previous script 💭  

So I think it's correct to not show an error for this case, what do you think? 🙏

Reply:
@Tuubz @kaungkhantsoe according to [the documentation](https://developer.android.com/studio/build/configure-app-module#set-application-id), `It must have at least two segments (one or more dots).` 😄

Reply:
I see. Then it is okay 🚀